modApiServer/src/org/aion/api/server/rpc3/types/RPCTypesConverter.java

```diff
@@ -0,0 +1,378 @@
+            try{
+                if(obj==null) return null;
+                JSONObject jsonObject = new JSONObject();
+                jsonObject.put("id", IntegerConverter.encode(obj.id));
+                jsonObject.put("method", StringConverter.encode(obj.method));
+                jsonObject.put("params", StringConverter.encode(obj.params));
+                jsonObject.put("jsonRPC", VersionTypeConverter.encode(obj.jsonRPC));
+                return jsonObject.toString();
+            }
+            catch (Exception e){
+                throw new ParseErrorRPCException();
+            }
+        }
+
+    }
+
+    public static class DataHexStringConverter{
+        private static final Pattern regex = Pattern.compile("^0x([0-9a-fA-F][0-9a-fA-F])+");
+
+        public static String decode(Object object){
+            try{
+                if (object!=null && checkConstraints(object.toString())){
+                    return StringConverter.decode(object);
+                }
+                else{
+                    throw new ParseErrorRPCException();
+                }
+            } catch(Exception e){
+                throw new ParseErrorRPCException();
+            }
+        }
+
+        public static String encode(String obj){
+            if (obj != null){
+                String result = StringConverter.encode(obj);
+                if(checkConstraints(result))
+                    return result;
+                else
+                    throw new ParseErrorRPCException();
+            }
+            else{
+                throw new ParseErrorRPCException();
+            }
+        }
+
+        private static boolean checkConstraints(String s){
+            return regex.matcher(s).find() && s.length() >= 4 && s.length() <= 2147483647;
+        }
+    }
+
+    public static class HexStringConverter{
+        private static final Pattern regex = Pattern.compile("^0x[0-9a-fA-F]+");
+
+        public static BigInteger decode(Object object){
+            try{
+                if (object!=null && checkConstraints(object.toString())){
+                    return BigIntegerConverter.decode(object);
+                }
+                else{
+                    throw new ParseErrorRPCException();
+                }
+            } catch(Exception e){
+                throw new ParseErrorRPCException();
+            }
+        }
+
+        public static String encode(BigInteger obj){
+            if (obj != null){
+                String result = BigIntegerConverter.encodeHex(obj);
+                if(checkConstraints(result))
+                    return result;
+                else
+                    throw new ParseErrorRPCException();
+            }
+            else{
+                throw new ParseErrorRPCException();
+            }
+        }
+
+        private static boolean checkConstraints(String s){
+            return regex.matcher(s).find() && s.length() >= 3 && s.length() <= 2147483647;
```

Comment:
use `Integer.MAX_VALUE` instead of the big number

Reply:
Define `PREFIX_LENGTH=2` and then use PREFIX_LENGHT + 1 and PREFIX_LENGHT + `Integer.MAX_VALUE` ？

Reply:
That number is Integer.MAX_VALUE it is just a consequence of how the processor was written. I will look into fixing it.